lib/elasticpress-integration.php

```diff
@@ -49,3 +53,34 @@ function sign_wp_request( array $args, string $url ) : array {
 	}
 	return $args;
 }
+
+
+function log_remote_request_errors( array $request ) {
+	if ( is_wp_error( $request['request'] ) ) {
+		trigger_error( sprintf( 'Error in ElasticPress request: %s (%s)', $request['request']->get_error_message(), $request['request']->get_error_code() ), E_USER_WARNING );
+	}
+}
+
+/**
+ * Default ElasticPress functionality is to fall-back to MySQL search when queries fail. We want to instead
+ * no-op the query when this happens, as we don't want to put lots of load on to MySQL.
+ *
+ * @param string $request
+ * @param WP_Query $query
+ * @return string
+ */
+function noop_wp_query_on_failed_ep_request( string $request, WP_Query $query ) : string {
+	if ( ! isset( $query->elasticsearch_success ) || $query->elasticsearch_success === true ) {
+		return $request;
+	}
+
+	global $wpdb;
+	return "SELECT * FROM $wpdb->posts WHERE 1=0";
```

Comment:
to clarify, this will return nothing, yea? We're short-circuiting the query?

Reply:
Correct, this was copied from elasticpress, as it does this too.